lib/subproviders/geth_api_double.js

```diff
@@ -139,23 +139,24 @@ GethApiDouble.prototype.eth_getBlockByNumber = function(block_number, include_fu
     }
 
     callback(null, {
-      number: to.hex(block.header.number),
+      number: to.hexWithoutLeadingZeroes(block.header.number),
       hash: to.hex(block.hash()),
       parentHash: to.hex(block.header.parentHash),
-      nonce: to.hex(block.header.nonce),
+      nonce: to.hexWithLeftPadding(block.header.nonce, 16),
       sha3Uncles: to.hex(block.header.uncleHash),
       logsBloom: to.hex(block.header.bloom),
       transactionsRoot: to.hex(block.header.transactionsTrie),
       stateRoot: to.hex(block.header.stateRoot),
-      receiptRoot: to.hex(block.header.receiptTrie),
+      receiptsRoot: to.hex(block.header.receiptTrie),
       miner: to.hex(block.header.coinbase),
-      difficulty: to.hex(block.header.difficulty),
-      totalDifficulty: to.hex(block.header.difficulty), // TODO: Figure out what to do here.
-      extraData: to.hex(block.header.extraData),
-      size: to.hex(1000), // TODO: Do something better here
-      gasLimit: to.hex(block.header.gasLimit),
-      gasUsed: to.hex(block.header.gasUsed),
-      timestamp: to.hex(block.header.timestamp),
+      difficulty: to.hexWithoutLeadingZeroes(block.header.difficulty),
+      totalDifficulty: to.hexWithoutLeadingZeroes(block.header.difficulty), // TODO: Figure out what to do here.
+      extraData: to.hexWithEvenLength(block.header.extraData),
+      mixHash: to.hex(block.header.mixHash),
```

Comment:
@mjkoo I don't see a `mixHash` field here: https://github.com/ethereum/wiki/wiki/JSON-RPC#returns-26

Reply:
Hmm yes you are right, that change was made due to a block deserialization issue in go-ethereum due to this field here https://github.com/ethereum/go-ethereum/blob/master/core/types/block.go#L84 but does not appear in the spec, so this may be a go-ethereum issue.